dataservice/api/sample_relationship/models.py

```diff
@@ -0,0 +1,165 @@
+from sqlalchemy import event, or_
+
+
+from dataservice.extensions import db
+from dataservice.api.common.model import Base, KfId
+from dataservice.api.sample.models import Sample
+
+
+class SampleRelationship(db.Model, Base):
+    """
+    Represents a relationship between two samples.
+
+    The relationship table represents a tree.
+
+    :param kf_id: Primary key given by the Kid's First DCC
+    :param created_at: Time of object creation
+    :param modified_at: Last time of object modification
+    :param parent_id: Kids first id of the parent Sample in the
+    relationship
+    :param child_id: Kids first id of the child Sample
+    in the relationship
+    :param external_parent_id: Name given to parent sample by contributor
+    :param external_child_id: Name given to child sample by contributor
```

Comment:
Why is this being populated in the relationship table? shouldn't this just be coming from sample?

Reply:
It is a bit redundant but I think its a tradeoff with ease of implementation + complexity of the query when fetching/searching for relationships with the external sample IDs.

If we didn't have these additional columns on the relationship table, then in order to get all relationships or search for specific relationships using the external parent or child sample IDs, we'd have to join the relationships table with the sample table on both the parent_id and child_id FK columns. 

We'd also have to break the pattern we follow in the schema layer of the API in order to return the parent Sample.external_id and the child Sample.external_id along with each sample relationship. We _could_ do this but I didn't think it was worth figuring this out with the old documentation.